### ISSUE TITLE
features/completion-suggestions differs from pressing TAB in subtle details

### ISSUE THREAD

**Issue by vaeth** [zsh-users/zsh-autosuggestions#342]
With features/completion-suggestions and strategy `completion`, the shown completion differs from that which is obtained by actually pressing TAB.

For instance, if /home is the only directory starting with /ho, then after `cd /ho` the suggested completion is `cd /home` while when pressing TAB would lead to 'cd /home/` (with a trailing slash). It would be much more convenient and less confusing to get the same result as with TAB.

**Comment by vaeth** [comment]
It would probably also be helpful there would not be a longer suggestion shown than that which would be given by TAB (this even means __no__ suggestion in the non-unique case).

Indeed, it is very confusing if after typing `ls` one receives the suggestion `lsblk`, or even worse after typing `xma` one receives the suggestion `xmame.SDL` (when also `xmame.x11` and even `xmame` are binaries matching the same beginning), or if after typing `ls /usr/bin/` one receives the suggestion `ls /usr/bin/x11`.

Especially if one is used to typing TAB frequently (or using auto-fu.zsh), this gives the wrong intuition that `ls` is actually not a complete command, that `xmame.SDL` is the only binary starting with `xma` (in particular, there is no `xmame`), or that `x11` is the _only_  binary existing in `/usr/bin/`, respectively. (Note that x11 is not even the alphabetically first file there; in fact, the choice in the non-unique case seems to be random or depending on the disk hash function.)

**Comment by ericfreese** [comment]
I think #350 addresses all of your concerns here. Take a look when you can.

**Comment by vaeth** [comment]
The “cd /ho`me/`” is now working, and also “/usr/bin/` `” is fine.
However, there is still “ls`blk`” and “/usr/bin/xma`me.SDL`” (despite xmame exists).
The latter is perhaps correct, because pressing 2 times tab shows indeed xmame.SDL before xmame in the completion list for some strange reason.

**Comment by ericfreese** [comment]
> However, there is still “lsblk” and “/usr/bin/xmame.SDL”

Strange- this is fixed for me. When I type `ls` I do not get a suggestion. After typing a `b`, I do get a suggestion for `lsblk`.

Pressing tab when the buffer has just `ls` gives these results:

```
ls          lsblk       lsinitcpio  lslocks     lsmem       lsns        lspcmcia    lsusb.py    
lsattr      lscpu       lsipc       lslogins    lsmod       lspci       lsusb  
```

Note `ls` is first, so there is no additional suggestion shown.

I wonder if you have some option/zstyle set that is affecting the behavior for you. A couple questions for you:

- What do you get when you press tab with just `ls` in the edit buffer (assuming you have zsh configured to list completions)?
- What happens if you run `zsh -f`, initialize `compinit`, and source the plugin manually? Do you get different behavior?

**Comment by vaeth** [comment]
The output was differently sorted for me. Binary searching in my .zshrc gave that the culprit was the line
```
zstyle ':completion:*' sort false
```
which is now rather unsurprising.

I am wondering why auto-fu.zsh does not extend ls to lsblk even with the above line.

**Comment by ericfreese** [comment]
I believe this ticket can be closed with #350 being merged to `develop`
